src/defaults.mligo

```diff
@@ -4,19 +4,26 @@
 #include "types.mligo"
 #include "proposal.mligo"
 
-let default_config (data : initial_config_data) : config = {
-  proposal_check = (fun (params, extras : propose_params * contract_extra) -> true);
-  rejected_proposal_return_value = (fun (proposal, extras : proposal * contract_extra) -> 0n);
-  decision_lambda = (fun (proposal, extras : proposal * contract_extra) -> (([] : (operation list)), extras));
-
-  quorum_threshold = data.quorum_threshold;
-  fixed_proposal_fee_in_token = data.fixed_proposal_fee_in_token;
-  voting_period = data.voting_period;
-  max_proposals = 500n;
-  max_votes = 1000n;
-
-  custom_entrypoints = (Big_map.empty : custom_entrypoints);
-}
+let default_config (data : initial_config_data) : config =
+  if    data.proposal_expired_time > data.proposal_flush_time
+     && data.proposal_flush_time > (data.voting_period.length * 2n)
+  then {
+    proposal_check = (fun (params, extras : propose_params * contract_extra) -> true);
+    rejected_proposal_return_value = (fun (proposal, extras : proposal * contract_extra) -> 0n);
+    decision_lambda = (fun (proposal, extras : proposal * contract_extra) -> (([] : (operation list)), extras));
+
+    quorum_threshold = data.quorum_threshold;
+    fixed_proposal_fee_in_token = data.fixed_proposal_fee_in_token;
+    voting_period = data.voting_period;
+    max_proposals = 500n;
+    max_votes = 1000n;
+    proposal_flush_time = data.proposal_flush_time;
+    proposal_expired_time = data.proposal_expired_time;
+
+    custom_entrypoints = (Big_map.empty : custom_entrypoints);
+  }
+  else
+    (failwith("INVALID_CONFIG") : config)
```

Comment:
Two small suggestions here, I think we can move the failing validation to a separate function (I intend to extend it in other issues too), plus considering that the error message is only to be evaluated for storage (and it will never be actually part of the contract `code`) we can have longer and more esplicative error messages.

My suggestion:
```suggestion
let validate_default_config (data : initial_config_data) : unit =
  if data.proposal_expired_time <= data.proposal_flush_time then
    failwith("proposal_expired_time needs to be bigger than proposal_flush_time")
  else if data.proposal_flush_time <= (data.voting_period.length * 2n) then
    failwith("proposal_flush_time needs to be more than twice the voting_period length")
  else unit

let default_config (data : initial_config_data) : config =
  let u : unit = validate_default_config(data) in {
    proposal_check = (fun (params, extras : propose_params * contract_extra) -> true);
    rejected_proposal_return_value = (fun (proposal, extras : proposal * contract_extra) -> 0n);
    decision_lambda = (fun (proposal, extras : proposal * contract_extra) -> (([] : (operation list)), extras));

    quorum_threshold = data.quorum_threshold;
    fixed_proposal_fee_in_token = data.fixed_proposal_fee_in_token;
    voting_period = data.voting_period;
    max_proposals = 500n;
    max_votes = 1000n;
    proposal_flush_time = data.proposal_flush_time;
    proposal_expired_time = data.proposal_expired_time;

    custom_entrypoints = (Big_map.empty : custom_entrypoints);
  }
```

Reply:
Definitely looks better, I've updated this now.